src/Microsoft.DotNet.Build.CloudTestTasks/UploadToAzure.cs

```diff
@@ -1,96 +1,229 @@
-                if (string.IsNullOrEmpty(relativeBlobPath))
+                throw new ArgumentException("No items were provided for upload.");
+            }
+
+            // first check what blobs are present
+            var checkListUrl = string.Format(
+                "https://{0}.blob.core.windows.net/{1}?restype=container&comp=list", 
+                this.AccountName, 
+                this.ContainerName);
+            var request = (HttpWebRequest)WebRequest.Create(checkListUrl);
+            request.Method = "GET";
+            var dt = DateTime.UtcNow;
+            request.Headers["x-ms-date"] = dt.ToString("R", CultureInfo.InvariantCulture);
+            request.Headers["x-ms-version"] = AzureHelper.StorageApiVersion;
+            request.Headers["Authorization"] = AzureHelper.AuthorizationHeader(
+                this.AccountName, 
+                this.AccountKey, 
+                "GET", 
+                dt, 
+                request);
+
+            this.Log.LogMessage(MessageImportance.High, "Sending request to check whether Container blobs exist");
+            var blobsPresent = new HashSet<string>();
+
+            try
+            {
+                XmlDocument doc;
+                using (var response = await request.GetResponseAsync())
                 {
-                    Log.LogError(string.Format("Metadata 'RelativeBlobPath' is missing for item '{0}'.", item.ItemSpec));
-                    result = false;
-                    loopState.Stop();
+                    doc = new XmlDocument();
+                    doc.LoadXml(new StreamReader(response.GetResponseStream()).ReadToEnd());
                 }
 
-                CloudBlockBlob blockBlob = container.GetBlockBlobReference(relativeBlobPath);
+                var nodes = doc.DocumentElement.GetElementsByTagName("Blob");
 
-                if (!Overwrite && blockBlob.Exists())
+                foreach (XmlNode node in nodes)
                 {
-                    Log.LogError(string.Format("The blob '{0}' already exists.", blockBlob.Uri));
-                    result = false;
-                    loopState.Stop();
+                    blobsPresent.Add(node["Name"].InnerText);
                 }
-
-                using (var fileStream = File.OpenRead(item.ItemSpec))
-                {
-                    Log.LogMessage(MessageImportance.Low, "Uploading {0} to {1}.", item.ItemSpec, blockBlob.Uri);
-                    blockBlob.UploadFromStream(fileStream);
-                }
-            });
+            }
+            catch (WebException e)
+            {
+                this.Log.LogError(e.Message);
+                this.Log.LogError(e.Status.ToString());
+                var resp = new StreamReader(e.Response.GetResponseStream()).ReadToEnd();
+                this.Log.LogError(resp);
+            }
+
+            this.Log.LogMessage(MessageImportance.High, "Received response to check whether Container blobs exist");
```

Comment:
"High" is really high. In general, a task should only have one "High" level event, saying the task started. The rest should probably be "Normal" or "Low", so that when building with "minimal" you still get a sense of progress, but messages that you can't really act on (like this one) don't shove off other messages.